src/Lecture1.hs

```diff
@@ -104,7 +109,8 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum str = sum $ map (\x -> read x :: Int) $ words str
```

Comment:
I see lots of GHC warnings in the module because you haven't written top-level type signatures for functions.

If you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
strSum str = sum $ map read $ words str
```